services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -755,14 +754,15 @@ protected void initDasCustody() {
             Duration.ofMinutes(5),
             configEip7594.getNumberOfColumns());
 
-    dasCustodySync = new DasCustodySync(dataColumnSidecarCustodyImpl, recoveringSidecarRetriever);
+    dasCustodySync = new DasCustodySync(custody, recoveringSidecarRetriever);
     eventChannels.subscribe(SlotEventsChannel.class, dasCustodySync);
 
     if (beaconConfig.p2pConfig().isDasLossySamplerEnabled()) {
       LOG.info("Lossy Sampler is not supported, starting basic sampler");
     }
     final DasSamplerBasic dasSampler =
-        new DasSamplerBasic(spec, dbAccessor, custody, nodeId, totalMyCustodySubnets);
+        new DasSamplerBasic(
+            spec, dbAccessor, custody, recoveringSidecarRetriever, nodeId, totalMyCustodySubnets);
```

Comment:
So sampler is switched from longPollCustody to custody. Why?

Reply:
Because it now has unconditional 5 sec timeout. It would else slow down the sync drastically. 
I'm planning to switch it back to `longPollCustody` after altering it to wait for the latest columns only